NineChronicles.Headless/GraphTypes/States/StateContext.cs

```diff
@@ -1,21 +1,30 @@
 using System.Collections.Generic;
+using System.Collections.Immutable;
 using Bencodex.Types;
 using Libplanet;
 using Libplanet.Action;
 using Libplanet.Assets;
 
 namespace NineChronicles.Headless.GraphTypes.States
 {
-    public class StateContext
+    public class StateContext : IAccountStateDelta
     {
-        public StateContext(AccountStateGetter accountStateGetter, AccountBalanceGetter accountBalanceGetter)
+        public StateContext(AccountStateGetter accountStateGetter, AccountBalanceGetter accountBalanceGetter, long chainTipIndex)
         {
             AccountStateGetter = accountStateGetter;
             AccountBalanceGetter = accountBalanceGetter;
+            ChainTipIndex = chainTipIndex;
```

Comment:
Where is chainTipIndex used? I can't see where it's used, is there a reason it was added?

Reply:
It looks like I didn't get everything over to this branch previously.  I needed the index to determine if 0 tickets is actually 0 or should be 8 and they haven't battled since the reset.  I have now included the code that uses the index.